core/src/main/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbe.java

```diff
@@ -0,0 +1,212 @@
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.LogCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.diff.DiffEntry;
+import org.eclipse.jgit.diff.DiffFormatter;
+import org.eclipse.jgit.errors.CorruptObjectException;
+import org.eclipse.jgit.errors.IncorrectObjectTypeException;
+import org.eclipse.jgit.errors.MissingObjectException;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.ObjectReader;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevTree;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.treewalk.CanonicalTreeParser;
+import org.eclipse.jgit.treewalk.TreeWalk;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * Using the analysis done by {@link SCMLinkValidationProbe},
+ * this probe will find number of unreleased commits in a repository
+ */
+@Component
+@Order(value = HasUnreleasedProductionChangesProbe.ORDER)
+public class HasUnreleasedProductionChangesProbe  extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HasUnreleasedProductionChangesProbe.class);
+
+    public static final int ORDER = LastCommitDateProbe.ORDER + 100;
+    public static final String KEY = "unreleased-production-changes";
+
+    @Override
+    public ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        List<String> productionPathsToCheckForCommits = new ArrayList<>();
+        Set<String> filesModifiedAfterRelease  = new HashSet<>();
+
+        final Matcher matcher = SCMLinkValidationProbe.GH_PATTERN.matcher(plugin.getScm());
+        if (!matcher.find()) {
+            return ProbeResult.failure(key(), "The SCM link is not valid");
+        }
+        final String folder = matcher.group("folder");
+
+        try (Git git = Git.init().setDirectory(context.getScmRepository().toFile()).call()) {
+            final LogCommand logCommand = git.log();
+
+            if (folder != null) {
+                productionPathsToCheckForCommits.add(folder + "pom.xml");
+                productionPathsToCheckForCommits.add(folder + "src/main");
+            }
+
+            productionPathsToCheckForCommits.add("pom.xml");
+            productionPathsToCheckForCommits.add("src/main");
+
+            productionPathsToCheckForCommits.forEach(logCommand:: addPath);
+            Iterable<RevCommit> commits = logCommand.call();
+
+            for (RevCommit commit : commits) {
+                Instant instant = Instant.ofEpochSecond(commit.getCommitTime());
```

Comment:
Is it a method on commit to get the instant directly?

Reply:
`commit.getCommitTime()` gives the epoch timestamp when the commit was made.

Reply:
`commit.getCommitterIdent().getWhenAsInstant()`